packages/protocol/contracts/requesters/RrpBeaconServer.sol

```diff
@@ -38,53 +38,52 @@ contract RrpBeaconServer is
     /// @param airnodeRrp_ Airnode RRP address
     /// @param metaAdmin_ Initial metaAdmin
     constructor(address airnodeRrp_, address metaAdmin_)
-        AirnodeRrpClient(airnodeRrp_)
+        AirnodeRrpRequester(airnodeRrp_)
         MetaAdminnable(metaAdmin_)
     {}
 
     /// @notice Called to request a beacon to be updated
     /// @dev Anyone can request a beacon to be updated. This is because it is
-    /// assumed that a beacon update request is always desirable and the
-    /// requester is paying for all the gas cost.
-    /// The requester must endorse both the caller of this function, and this
-    /// very AirnodeRrpBeaconServer contract for the Airnode to fulfill this
-    /// request.
+    /// assumed that a beacon update request is always desirable, and the
+    /// requester and sponsor will pay for the gas cost.
+    /// The sponsor must sponsor both the caller of this function, and this
+    /// very RrpBeaconServer contract for the Airnode to fulfill this request.
     /// The template used here must specify a single point of data of type
     /// `int256` to be returned (because this is what `fulfill()` expects).
     /// @param templateId Template ID of the beacon to be updated
-    /// @param requester Requester whose designated wallet will be used to
-    /// fulfill this request
-    /// @param designatedWallet Designated wallet that will be used to fulfill
-    /// this request
+    /// @param sponsor Sponsor whose wallet will be used to fulfill this
+    /// request
+    /// @param sponsorWallet Sponsor wallet that will be used to fulfill this
+    /// request
     function requestBeaconUpdate(
         bytes32 templateId,
-        address requester,
-        address designatedWallet
+        address sponsor,
+        address sponsorWallet
     ) external override {
         // Note that AirnodeRrp will also check if the requester has endorsed
-        // this AirnodeRrpBeaconServer in the `makeRequest()` call
+        // this RrpBeaconServer in the `makeRequest()` call
         require(
-            airnodeRrp.requesterToClientAddressToEndorsementStatus(
-                requester,
+            airnodeRrp.sponsorToRequesterToSponsorshipStatus(
+                sponsor,
                 msg.sender
             ),
-            "Caller not endorsed by requester"
+            "Caller not sponsored"
         );
-        bytes32 requestId = airnodeRrp.makeRequest(
+        bytes32 requestId = airnodeRrp.makeTemplateRequest(
             templateId,
-            requester,
-            designatedWallet,
+            sponsor,
+            sponsorWallet,
             address(this),
             this.fulfill.selector,
-            "0x"
+            ""
```

Comment:
Here's another bug. `"0x"` is an empty `bytes` when making ethers calls, in Solidity and empty `bytes` is `""`.